packages/nocms-plugin-directory-resources/src/renderDirectoryResourceJs.js

```diff
@@ -0,0 +1,46 @@
+import { rollup  } from 'rollup';
+import fs from 'fs';
+import path from 'path';
+import rollupDefaults from './rollupDefaults';
+
+let cache;
+
+async function rollupWithCache(options) {
+	const bundle = await rollup({
+		cache,
+		...options
+	});
+	cache = bundle.cache;
+	return bundle;
+}
+
+/**
+ *
+ * @param {*} directoryResource
+ */
+export default async function renderDirectoryResourceJs(directoryResource) {
+	const jsPaths = [
+		path.join(directoryResource.physicalPath, '_index.js'),
+		path.join(directoryResource.physicalPath, 'index.js')
+	];
+
+	const jsPath = jsPaths.find(path => fs.existsSync(path));
```

Comment:
In general I suggest using promisify and avoid blocking `sync` functions, but I am not sure it makes a difference here.

Reply:
Good point, I think the worker is "stuck" in that one until the file is found anyways, will keep an eye out for that though

Reply:
Not going to make a difference here as primary constraint is CPU and not IO